### PR TITLE
Adding a few event handlers and a bit of focus management.

### PR DESCRIPTION
**Pull request by patrickfournier**

### CODE REVIEW

File: Assets/Editor/E05_Controls.cs [hunk 1]
```diff
@@ -77,6 +101,9 @@ void AddTask()
             {
                 m_TasksContainer.contentView.AddChild(CreateTask(m_TextField.text));
                 m_TextField.text = "";
+
+                // Give focus back to text field.
+                m_TasksContainer.focusController.SwitchFocus(m_TextField);
```
Comment (Antoine-Lassauzay) [inline Assets/Editor/E05_Controls.cs]: Should we make Focus controller public? Isn't here an extension method to give focus to an object?

Comment (patrickfournier) [inline Assets/Editor/E05_Controls.cs]: Not sure to understand what you mean here. There is one focus controller per panel. myVisualElement.focusControler is a shortcut for myVisualElement.panel.focusController.

Comment (Antoine-Lassauzay) [inline Assets/Editor/E05_Controls.cs]: Right, I'm just wondering if would be easier to myVisualElement.TakeFocus(). But your approach is fine ; just debating if FocusController is a class meant to be public / documented etc.

Comment (patrickfournier) [inline Assets/Editor/E05_Controls.cs]: In fact you are right we could also write m_TextField.Focus() here. It is not an extension method, it is part of the Focusable class.
However we need FocusController to be public to get access to the currently focused element.

**Comment by patrickfournier** [comment]
trunk is ready for this PR

**Pull request merged**